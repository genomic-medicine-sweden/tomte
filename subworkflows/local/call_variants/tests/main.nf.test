nextflow_workflow {

    name "Test Workflow CALL_VARIANTS"
    script "subworkflows/local/call_variants/main.nf"
    workflow "CALL_VARIANTS"
    config "./nextflow.config"
    tag "call_variants"

    setup {
        run("GUNZIP", alias: "GUNZIP_FASTA") {
        script "../../../../modules/nf-core/gunzip/main.nf"
        process {
            """
            input[0] = Channel.of([
                [ id:'fasta' ],
                file("${projectDir}/test_data/grch37_chr21.fa.gz", checkIfExists: true)
            ])
            """
            }
        }

        run("SAMTOOLS_FAIDX") {
        script "../../../../modules/nf-core/samtools/faidx/main.nf"
        process {
            """
            input[0] = GUNZIP_FASTA.out.gunzip
            input[1] = [[],[]]
            """
            }
        }

        run("GATK4_CREATESEQUENCEDICTIONARY", alias: "BUILD_DICT") {
        script "../../../../modules/nf-core/gatk4/createsequencedictionary/main.nf"
        process {
            """
            input[0] = GUNZIP_FASTA.out.gunzip
            """
            }
        }
         
    }

    test("bcftools") {
        when {
            workflow {
                """
                // Channel for the BAM sample
                ch_bam = Channel.of([
                        [ case:'finequagga', id:'ACC5963A1', sample:'ACC5963A1', strandedness:'reverse', sex:'NA', single_end:false, fq_pairs:1, is_fastq:false ],
                        file("${projectDir}/test_data/ACC5963A1.bam", checkIfExists: true),
                        file("${projectDir}/test_data/ACC5963A1.bam.bai", checkIfExists: true)
                    ])
                input[0] = ch_bam
                input[1] = GUNZIP_FASTA.out.gunzip.collect()
                input[2] = SAMTOOLS_FAIDX.out.fai.collect()
                input[3] = BUILD_DICT.out.dict.collect()
                input[4] = 'bcftools'
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcf.collect{path(it[1]).vcf.getVariantsAsStrings(2)},
                    file(workflow.out.tbi.get(0).get(1)).name,
                    workflow.out.stats,
                    workflow.out.versions,
                ).match()}
            )
        }
    }

    test("gatk") {
        
        when {
            workflow {
                """
                // Channel for the BAM sample
                ch_bam = Channel.of([
                        [ case:'finequagga', id:'ACC5963A1', sample:'ACC5963A1', strandedness:'reverse', sex:'NA', single_end:false, fq_pairs:1, is_fastq:false ],
                        file("${projectDir}/test_data/ACC5963A1.bam", checkIfExists: true),
                        file("${projectDir}/test_data/ACC5963A1.bam.bai", checkIfExists: true)
                    ])
                input[0] = ch_bam
                input[1] = GUNZIP_FASTA.out.gunzip.collect()
                input[2] = SAMTOOLS_FAIDX.out.fai.collect()
                input[3] = BUILD_DICT.out.dict.collect()
                input[4] = 'gatk'
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcf.collect{path(it[1]).vcf.getVariantsAsStrings(2)},
                    file(workflow.out.tbi.get(0).get(1)).name,
                    workflow.out.stats,
                    workflow.out.versions,
                ).match()}
            )
        }
    }
}
