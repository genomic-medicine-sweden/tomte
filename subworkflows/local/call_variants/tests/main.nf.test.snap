{
    "gatk": {
        "content": [
            [
                "VcfFile [chromosomes=[chr21], sampleCount=1, variantCount=2502, phased=false, phasedAutodetect=false]"
            ],
            [
                "VcfFile [chromosomes=[chr21], sampleCount=1, variantCount=2502, phased=false, phasedAutodetect=false]"
            ],
            "ACC5963A1_split_rmdup_info.vcf.gz.tbi",
            [
                [
                    {
                        "case": "finequagga",
                        "id": "ACC5963A1",
                        "sample": "ACC5963A1",
                        "strandedness": "reverse",
                        "sex": "NA",
                        "single_end": false,
                        "fq_pairs": 1,
                        "is_fastq": false
                    },
                    "ACC5963A1.bcftools_stats.txt:md5,09a5b7beb572c5754f44f2dea20024a2"
                ]
            ],
            [
                "versions.yml:md5,0aa84c73aee35f1d02fe6b4d1bdffde2",
                "versions.yml:md5,4c5bc1efe43ee5d2522088d5e6547a9a",
                "versions.yml:md5,59d806d3927eedb15e480b9e6a814326",
                "versions.yml:md5,79a201dd2139b6ac9fbb55f5c9b3d2ca"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-11T11:11:42.578799"
    },
    "bcftools": {
        "content": [
            [
                "VcfFile [chromosomes=[chr21], sampleCount=1, variantCount=805, phased=false, phasedAutodetect=false]"
            ],
            [
                "VcfFile [chromosomes=[chr21], sampleCount=1, variantCount=805, phased=false, phasedAutodetect=false]"
            ],
            "ACC5963A1_split_rmdup_info.vcf.gz.tbi",
            [
                [
                    {
                        "case": "finequagga",
                        "id": "ACC5963A1",
                        "sample": "ACC5963A1",
                        "strandedness": "reverse",
                        "sex": "NA",
                        "single_end": false,
                        "fq_pairs": 1,
                        "is_fastq": false
                    },
                    "ACC5963A1.bcftools_stats.txt:md5,3b763dea64d311e1689cf52a72b33e44"
                ]
            ],
            [
                "versions.yml:md5,4c5bc1efe43ee5d2522088d5e6547a9a",
                "versions.yml:md5,59d806d3927eedb15e480b9e6a814326",
                "versions.yml:md5,79a201dd2139b6ac9fbb55f5c9b3d2ca",
                "versions.yml:md5,8454ade6be56bd4f547e2b9b836fd947"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-11T11:10:14.823748"
    }
}